packages/common/src/content/contentUtils.ts

```diff
@@ -437,3 +440,52 @@ const getContentRelativeUrl = (
     content.typeKeywords
   );
 };
+
+// Tests can be found in packages/common/test/content/content.test.ts
+/**
+ * Get the status of a content item
+ * @param entity the content item
+ * @returns the status of the content item
+ */
+export async function getContentStatus(
+  entity: IHubEditableContent,
+  requestOptions: IRequestOptions
+): Promise<IHubServiceBackedContentStatus> {
+  const unavailable = {
+    kind: "service",
+    service: {
+      availability: "unavailable",
+    },
+  } as IHubServiceBackedContentStatus;
+
+  if (entity.url) {
+    // set up our two promises: one to get the service definition and one to sleep for 3 seconds
+    const definitionPromise: Promise<IFeatureServiceDefinition> = getService({
+      url: entity.url,
+      authentication: requestOptions.authentication,
+    });
+    const timeoutPromise = (ms: number) => {
+      return new Promise((resolve) => setTimeout(resolve, ms));
+    };
```

Comment:
Looks like there are a couple of places that could use this, so I think it is worth moving to it's own util and ideally updating those places to use it (or at least adding a comment that they should use it):

```bash
packages/common/src/downloads/_internal/file-url-fetchers/fetchExportItemDownloadFileUrl.ts:
  129      progressCallback && progressCallback(DownloadOperationStatus.PROCESSING);
  130:     await new Promise((resolve) => setTimeout(resolve, pollInterval));
  131      return pollForJobCompletion(

packages/common/src/downloads/_internal/file-url-fetchers/fetchHubApiDownloadFileUrl.ts:
  131    // Operation still in progress, poll again
  132:   await new Promise((resolve) => setTimeout(resolve, pollInterval));
  133    return pollDownloadApi(requestUrl, pollInterval, progressCallback);

packages/common/src/items/_internal/_wait-for-item-ready.ts:
  7  function delay(milliseconds: number) {
  8:   return new Promise((resolve) => setTimeout(resolve, milliseconds));
  9  }

packages/common/src/utils/poll.ts:
  33      const results = await new Promise<T>((resolve) => {
  34:       setTimeout(async () => {
  35          const res = await requestFn();
```

Reply:
After further investigation, this is wayyyy more repeated than what I realized. 

Made a spin off and assigned it to platform -- https://devtopia.esri.com/dc/hub/issues/10953

Reply:
Thanks for creating that issue, but it's not the sort of thing that a developer on any thread (least of all Platform) will have an easy time pulling into their sprint. Instead, it is the sort of thing that we should try to fold into our work, in little steps if needed. The first step, and the minimum amount of work is that you should do in this PR, is create the hub.js fn and use it in the new code you are introducing. You might be able to avoid writing a test for it by using it in at least one of the other places that already has coverage. If it's too big a lift to update all the other places, that's fine, but the next step is to at least add a comment to the other places to say that they should be refactored to use new fn. Finally, you can scope this to hub.js and hub-components; no need to update the Ember code at this point. 
So ideally this PR will:
- add the new fun
- use that fn in your code
- get coverage for that fn either through using in an existing fn w/ coverage or writing a test
- either replace or add comments to other hub.js locations

And then your PR that updates opendata-ui to use the fn you introduce in this PR would:
- either replace or add comments to the hub-components locations